admin/listPurchases/listView.html

```diff
@@ -295,25 +295,29 @@
     });
 }
 
-function requestOrders(callback) {
-    var req = new XMLHttpRequest();
-    req.open("GET", "/admin/orders");
-    req.setRequestHeader("Content-Type", "application/json");
-    req.onreadystatechange = function() {
-        if(req.readyState != 4) {
-            return;
-        }
-        if(req.status == 200) {
-            return callback(null, JSON.parse(this.responseText));
+async function requestOrders(callback) {
+    let response;
+    try{
+        response = await fetch("/admin/orders", {
+            method: 'GET',
+            headers: {
+                "Content-Type": "application/json",
+            },
+        });
+        if(!response.ok){
+            callback({
+                trace: new Error("Error response from server on request"),
+                status: response.status
+            })
         }
+    }
+    catch(error){
         callback({
-            trace: new Error("Error response from server on request"),
-            status: req.statusText,
-            responseType: req.responseType,
-            response: req.response
+            trace: new Error(`Request failed: ${error}`)
```

Comment:
This strips a lot of info about what has gone wrong by stringifying the error.

Instead:

```js
callback({
  trace: new Error("Request failed"),
  previous: error,
});
```

This preserves the full error and makes it explorable in the browser console.